_posts/2021-04-21-frequency-band-control-in-unetstack.md

```diff
@@ -0,0 +1,199 @@
+
+The usage and meaning of parameters can be found in [Command Reference](https://unetstack.net/handbook/unet-handbook.html#_command_reference)
+
+## FHBFSK frequency band control
+
+For FHBFSK scheme, the parameter `phy[3].fmin` specifies the starting frequency. `phy[3].fstep` specifies the frequency steps and  `phy[3].hops` the total number of frequencies used. 
+
+By default you may see in the Unet audio modem 
+```
+> phy[3]
+  ...
+  fmin = 9520.0
+  fstep = 160.0
+  hops = 13
+```
+
+So it starts from 9520 Hz and goes to 9520 + 160 x (13-1) x 2  = 13360 Hz as seen below. 
+The effective bandwidth can be roughly taken as fstep x hops x 2.
+
+
+<p align="center"><img src="../assets/img/freqBandControl/fhbfsk-psd.png" style="zoom:50%;"/></p>
+
+
+We can alter the above parameters to use a band anywhere between 6kHz and 18kHz, which is the computationally feasible band of the Unet audio modem. 
+
+Note that just as in OFDM, if there is a frequency band to be avoided, we can position the used band to the "left" or "right" of it by adjusting `fmin`, `fstep` and `hops`.
+
+
+## Control of Preamble bandwidth
+
+To complete the picture of bandwidth control, it must be noted that the signal that is transmitted not only has a data modulated portion, but also a detection preamble at the start. This is usually a pseudo random noise signal or m-sequences, hyperbolic chirp (HFM) etc that has good auto correlation properties. 
+
+[Preamble class](https://unetstack.net/javadoc/3.3/org/arl/yoda/Preamble.html) provides both the above types of preambles via the methods such as mseq(..), hfmUpSweep(..), hfmDownSweep(..).
+It allows controlling the bandwidth and centre frequency of the preamble.
+
+Let's look at one of them
+`public static Preamble hfmDownSweep(int len, int sps, float fc)`
+- `len` is the number of baseband samples
+- `sps` controls the bandwidth
+  Preamble bandwidth = baseband bandwidth/sps. Note that sps must be an integer.
+  Thus for Unet audio, sps=4 creates a bandwidth of 12kHz/4 = 3kHz.
+- `fc` controls the centre frequency. This is normalized to the baseband rate. So for Unet audio, `fc=0.25` will shift the carrier by 12kHz x 0.25 = 3kHz. i.e., from 12kHz to 15kHz. 
+
+For e.g., the following creates an HFM down sweep signal with 1000 baseband samples, bandwidth of 3kHz, centered at 15kHz.
+```
+> phy[3].preamble = org.arl.yoda.Preamble.hfmDownSweep(1000, 4, 0.25)
+(1000 samples)
+
+```
+To transmit this preamble alone (with no additional signal)
+
+```
+> phy << new TxBasebandSignalReq(preamble: 3)
+```
+
+In the diag scope, let us look at the passband PSD. With an appropriate trigger, you may get something as follows. It clearly shows the centre of 15kHz and the bandwidth of 3kHz.
+
+<p align="center"><img src="../assets/img/freqBandControl/preamble.png" style="zoom:50%;"/></p>
+
+The user can also generate and set custom preambles. 
```

Comment:
Details of custom preamble is available in section `16.4. Transmitting and detecting preambles` of the [unet handbook](https://unetstack.net/handbook/unet-handbook.html#_baseband_service).

Reply:
Could you add this link?

Reply:
Added.